packages/devtools_app/lib/src/primitives/utils.dart

```diff
@@ -1501,3 +1501,29 @@ bool isPrimativeInstanceKind(String? kind) {
 /// Returns the file name from a URI or path string, by splitting the [uri] at
 /// the directory separators '/', and returning the last element.
 String? fileNameFromUri(String? uri) => uri?.split('/').last;
+
+/// Calculates subtraction of two maps.
+///
+/// Result map keys is union of the imput maps' keys.
+///
+/// The function invokes:
+/// - `subtract` for values if keys presented in both input maps
+/// - `transform` for keys presented in just `minuend`
+/// - `negate` for keys presented in just `subtrahend`
+Map<K, V2> subtractMaps<K, V1, V2>({
+  required Map<K, V1>? minuend,
+  required Map<K, V1>? subtrahend,
```

Comment:
while these names are mathematically accurate, I wonder if it would be easier to read and understand this code if we used names like `subtract` and `from`, so the method call looked like: `final myNewMap = subtractMaps(subtract: map1, from: map2)`

Or even better an extension method on Map so it would look like:
`final myNewMap = map1.subtract(map2)`